lib_agc/src/agc.xc

```diff
@@ -576,17 +576,13 @@ static void agc_process_channel(agc_ch_state_t &state, vtb_ch_pair_t samples[AGC
 
         // Update near-end-activity timer
         if(vtb_gte_u32_u32(state.lc_near_power_est, vtb_mul_u32_u32(delta, state.lc_bg_power_est))){
-            if(state.lc_t_far == 0){
-                // Near-end speech only
+            if(state.lc_t_far == 0 || (state.lc_t_far && (state.lc_corr_factor < state.lc_corr_threshold.m))){
+                // Near-end speech only or Double talk
                 state.lc_t_near = state.lc_n_frame_near;
             }
-            else if(state.lc_t_far && (state.lc_corr_factor < state.lc_corr_threshold.m)){
-                // Double talk
-                state.lc_t_near = state.lc_n_frame_near>>1;
-            }
             else {
                 // Far-end speech only
-                state.lc_t_near = 0;
+                // Do nothing
```

Comment:
Either eliminate the `else` clause altogether or include in it an empty statement, i.e., just a `;`.